include/solvers/fluid_dynamics_matrix_free.h

```diff
@@ -99,15 +100,20 @@ class MFNavierStokesPreconditionGMG
    * @param[in] simulation_control Required to get the time stepping method.
    * @param[in] fe Describes the FE system for the vector-valued problem.
    */
-  MFNavierStokesPreconditionGMG(
-    const SimulationParameters<dim>          &simulation_parameters,
-    const DoFHandler<dim>                    &dof_handler,
-    const DoFHandler<dim>                    &dof_handler_fe_q_iso_q1,
-    const std::shared_ptr<Mapping<dim>>      &mapping,
-    const std::shared_ptr<Quadrature<dim>>   &cell_quadrature,
-    const std::shared_ptr<Function<dim>>      forcing_function,
-    const std::shared_ptr<SimulationControl> &simulation_control,
-    const std::shared_ptr<FESystem<dim>>      fe);
+  MFNavierStokesPreconditionGMGBase(
```

Comment:
The class hierarchy of the multigrid operators: `MFNavierStokesPreconditionGMGBase` -> `MFNavierStokesPreconditionGMG` -> `MFNavierStokesVANSPreconditionGMG`. The derivatives need to implement `create_level_operator()` and `initialize(...)`.